connector/saml/saml.go

```diff
@@ -263,6 +272,28 @@ func (p *provider) POSTData(s connector.Scopes, id string) (action, value string
 		return "", "", fmt.Errorf("marshal authn request: %v", err)
 	}
 
+	// Compress the request if redirect-binding is used
+	if p.redirectBinding {
+		var buf bytes.Buffer
+		zw, e := flate.NewWriter(&buf, flate.DefaultCompression)
+		if e == nil {
```

Comment:
here `e == nil` and `err != nil` is handled in one place below.

I use `e == nil` because the sequence will go through 3 steps: NewWriter, Write and Flush. Each may fail and should bail out immediately. So use `if e == nil { do...}` will make the code simpler, and handle the error in one place.